### PR TITLE
Improvements and correction to snow.loss_townsend

### LINKED ISSUE #1636: Corrections to Townsend snow model
Private communications with the model's author have turned up some issues with the pvlib implementation. Chief among the issues is  this part of the calculation:

```
    lower_edge_height_clipped = np.maximum(lower_edge_height, 0.01)
    gamma = (
        slant_height
        * effective_snow_weighted_m
        * cosd(surface_tilt)
        / (lower_edge_height_clipped**2 - effective_snow_weighted_m**2)
        * 2
        * tand(angle_of_repose)
    )

    ground_interference_term = 1 - C2 * np.exp(-gamma)
```

When `lower_edge_height_clipped` < `effective_snow_weighted_m`, `gamma` < 0 and the `ground_interference_term` can become negative. In contrast, the author's intent is that C2 < `ground_interference_terms` < 1. The author recommends clipping the squared difference (lower bound being worked out but will be something like 0.01.).

Other issues appear to arise from the unit conversions. The published model uses inches for distance and snow depth. The pvlib code uses cm for snow depth (convenience for working with external snow data) and m for distances (for consistency with the rest of pvlib). After several steps, including the `ground_interference_term` calculation, the code converts from cm or m to inches to apply the final formula for loss (since the formula involves some coefficients determined by a regression). It would be easier to trace the pvlib code back to the paper if the internal unit conversions (from cm / m to inches) were done earlier.

### PR DESCRIPTION
**Pull request by cwhanse**
- [x] Closes #1636 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Adds a lower bound to a term in the calculation of the ground interference term that prevents a negative squared distance from driving loss fractions to 1.0. Although the reference doesn't describe this limit, I consider its omission a bug in that results from the uncorrected function should be regarded as wrong.

Adds a new parameter, string_factor, that avoids loss fractions of 1.0 for systems with multiple strings across the slant height. This new parameter was added to the model by its author after publication of the reference.

### CODE REVIEW

**Comment by cwhanse** [comment]
@kanderso-nrel curious what you think should be done with the current test. If it is checking against output from the original function, that output is wrong. The new tests have limited precision so may not serve as suitable regression checks.

File: pvlib/tests/test_snow.py [hunk 1]
```diff
@@ -118,12 +121,92 @@ def test_loss_townsend():
...
+         )
+    ]
+)
+def test_loss_townsend_cases(poa_global, surface_tilt, slant_height,
+                             lower_edge_height, string_factor, expected):
+    # test cases from Townsend, 1/27/2023, addeed by cwh
+    # snow_total in inches, convert to cm for pvlib
+    snow_total = np.asarray(
+        [20, 15, 10, 4, 1.5, 0, 0, 0, 0, 1.5, 4, 15], dtype=float) * 2.54
+    # snow events are an overage for each month
```
Comment (kanderso-nrel) [inline pvlib/tests/test_snow.py:198]: "average", and maybe worth indicating this (that the number can be float, not just int) in the parameter's docstring description?

File: docs/sphinx/source/whatsnew/v0.9.5.rst [hunk 2]
```diff
@@ -16,11 +16,14 @@ Deprecations
 
 Enhancements
 ~~~~~~~~~~~~
-
+* Added the optional `string_factor` parameter to
+  :py:func:``pvlib.snow.loss_townsend`` (:issue:`1636`, :pull:`1653`)
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.9.5.rst:20]: `:py:func:`s need single backticks, not double

**Review by kanderso-nrel — APPROVED** [review]
LGTM.  Thanks @cwhanse both for the PR and the behind-the-scenes work on this one.

**Pull request merged**